kubernetes/monitoring/services/mongo/monit-mongo.yaml

```diff
@@ -0,0 +1,164 @@
+# [ConfigMap] [Service] [StatefulSet]
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: monit-mongo-configmap
+  namespace: monit-mongo
+data:
+  mongo.conf: |
+    storage:
+      dbPath: /data/db
+  ensure-users.js: |
+    const targetDbStr = 'test';
+    const rootUser = process.env["$MONGO_ROOT_USERNAME"];
+    const rootPass = process.env["MONGO_ROOT_PASSWORD"];
+    const usersStr = process.env["MONGO_USERS_LIST"];
+
+    const adminDb = db.getSiblingDB('admin');
+    adminDb.auth(rootUser, rootPass);
+    print('Successfully authenticated admin user');
+
+    const targetDb = db.getSiblingDB(targetDbStr);
+
+    const customRoles = adminDb
+      .getRoles({rolesInfo: 1, showBuiltinRoles: false})
+      .map(role => role.role)
+      .filter(Boolean);
+
+    usersStr
+      .trim()
+      .split(';')
+      .map(s => s.split(':'))
+      .forEach(user => {
+        const username = user[0];
+        const rolesStr = user[1];
+        const password = user[2];
+
+        if (!rolesStr || !password) {
+          return;
+        }
+
+        const roles = rolesStr.split(',');
+        const userDoc = {
+          user: username,
+          pwd: password,
+        };
+
+        userDoc.roles = roles.map(role => {
+          if (!~customRoles.indexOf(role)) {
+            return role;
+          }
+          return {role: role, db: 'admin'}; 
+        });
+
+        try {
+          targetDb.createUser(userDoc);
+        } catch (err) {
+          if (!~err.message.toLowerCase().indexOf('duplicate')) {
+            throw err;
+          }
+        }
+      });
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: monit-mongo
+  namespace: monit-mongo
+spec:
+  type: NodePort
+  ports:
+    - port: 27017
+      protocol: TCP
+      targetPort: 27017
+      nodePort: 32000
+  selector:
+    app: monit-mongo
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: monit-mongo
+  namespace: monit-mongo
+spec:
+  serviceName: monit-mongo
+  replicas: 1
+  selector:
+    matchLabels:
+      app: monit-mongo
+  template:
+    metadata:
+      namespace: monit-mongo
+      labels:
+        app: monit-mongo
+        selector: monit-mongo
+    spec:
```

Comment:
I suggest to add `livenessProbe` section which will ensure that k8s can ping service and restart it. Also, please add `resources` section which will constrain required resources for this service. You can view how we use them in any manifest files from cmsweb area, or consult with @muhammadimranfarooqi